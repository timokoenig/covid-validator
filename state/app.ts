import { entity, persistence } from 'simpler-state'

export type AppState = {
  country: string
  state: string
  purpose: string
  counter: number
  showCounter: boolean
}

export const app = entity<AppState>(
  { country: 'DE', state: '', purpose: '3G', counter: 0, showCounter: false },
  [persistence('app')]
)

export const setCountry = (country: string) => {
  app.set({ ...app.get(), country })
}

export const setState = (state: string) => {
  app.set({ ...app.get(), state })
}

export const setPurpose = (purpose: string) => {
  app.set({ ...app.get(), purpose })
}

export const toggleCounter = () => {
  const state = app.get()
  app.set({ ...state, showCounter: !state.showCounter })
}

export const resetCounter = () => {
  app.set({ ...app.get(), counter: 0 })
}

export const decreaseCounter = () => {
  const state = app.get()
  if (state.counter == 0) return
  app.set({ ...state, counter: state.counter - 1 })
}

export const increaseCounter = () => {
  const state = app.get()
  app.set({ ...state, counter: state.counter + 1 })
}
