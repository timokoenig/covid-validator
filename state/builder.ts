import { entity, persistence } from 'simpler-state'
import { CustomRule } from '../utils/certlogic'

export type BuilderState = {
  customRules: CustomRule[]
}

export const builder = entity<BuilderState>({ customRules: [] }, [persistence('builder')])

export const setRules = (customRules: CustomRule[]) => {
  builder.set({ ...builder.get(), customRules })
}
