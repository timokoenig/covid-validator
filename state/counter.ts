import { entity, persistence } from 'simpler-state'

export const counter = entity(0, [persistence('counter')])

export const reset = () => {
  counter.set(0)
}

export const decrease = () => {
  if (counter.get() == 0) return
  counter.set(counter.get() - 1)
}

export const increase = () => {
  counter.set(counter.get() + 1)
}
