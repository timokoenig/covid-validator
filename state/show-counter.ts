import { entity, persistence } from 'simpler-state'
import { reset } from './counter'

export const showCounter = entity(false, [persistence('show-counter')])

export const toggle = () => {
  reset()
  showCounter.set(!showCounter.get())
}
