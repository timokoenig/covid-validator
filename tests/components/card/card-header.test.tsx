import React from 'react'
import renderer from 'react-test-renderer'
import { resetAll } from 'simpler-state'
import CardHeader from '../../../components/card/card-header'
import { app } from '../../../state/app'

describe('CardHeader', () => {
  beforeEach(() => {
    resetAll()
  })

  it('renders correctly', () => {
    app.set({ ...app.get(), country: 'foo', state: 'bar' })

    const tree = renderer.create(<CardHeader />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
