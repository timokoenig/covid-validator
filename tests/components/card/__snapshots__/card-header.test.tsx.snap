// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardHeader renders correctly 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-16glbkg{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;background-color:blue.300;}",
      }
    }
    data-emotion="css 16glbkg"
  />,
  <div
    className="css-16glbkg"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-1a55jeh{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-appearance:none;-moz-appearance:none;-ms-appearance:none;appearance:none;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;position:relative;white-space:nowrap;vertical-align:middle;outline:2px solid transparent;outline-offset:2px;width:auto;-webkit-flex:1;-ms-flex:1;flex:1;background-color:inherit;padding-left:5px;padding-top:3px;padding-bottom:3px;height:auto;border-radius:0px;border-bottom:1px;border-bottom-color:blue.300;}.css-1a55jeh:hover,.css-1a55jeh[data-hover]{background-color:blue.400;}.css-1a55jeh:active,.css-1a55jeh[data-active]{background-color:blue.300;}",
        }
      }
      data-emotion="css 1a55jeh"
    />
    <button
      className="chakra-button css-1a55jeh"
      onClick={[Function]}
      type="button"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-70qvj9{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
          }
        }
        data-emotion="css 70qvj9"
      />
      <div
        className="css-70qvj9"
      >
        <span
          className="fi fi-foo fis"
          style={
            Object {
              "borderRadius": 12.5,
              "height": 25,
              "width": 25,
            }
          }
        />
      </div>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-ubxiqz{text-align:center;-webkit-flex:1;-ms-flex:1;flex:1;-webkit-margin-start:6px;margin-inline-start:6px;-webkit-margin-end:6px;margin-inline-end:6px;color:white;}",
          }
        }
        data-emotion="css ubxiqz"
      />
      <div
        className="css-ubxiqz"
      >
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-1rd7vmm{font-size:lg;word-break:break-word;white-space:normal;font-weight:bold;}",
            }
          }
          data-emotion="css 1rd7vmm"
        />
        <p
          className="chakra-text css-1rd7vmm"
        >
          n/a
        </p>
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-qsvu6k{font-size:md;word-break:break-word;white-space:normal;font-weight:400;}",
            }
          }
          data-emotion="css qsvu6k"
        />
        <p
          className="chakra-text css-qsvu6k"
        >
          
        </p>
      </div>
      <div
        className="css-70qvj9"
      >
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-1sb26sn{font-size:lg;justify-self:flex-end;font-weight:bold;color:white;}",
            }
          }
          data-emotion="css 1sb26sn"
        />
        <p
          className="chakra-text css-1sb26sn"
        >
          
        </p>
      </div>
    </button>
  </div>,
]
`;
